--- a/build/src/components/embedder_support/user_agent_utils.cc
+++ b/build/src/components/embedder_support/user_agent_utils.cc
@@ -269,9 +271,6 @@ std::string GetProductAndVersion(
 std::string GetUserAgentInternal(
     UserAgentReductionEnterprisePolicyState user_agent_reduction) {
   std::string product = GetProductAndVersion(user_agent_reduction);
-  if (base::CommandLine::ForCurrentProcess()->HasSwitch(kHeadless)) {
-    product.insert(0, "Headless");
-  }

 #if BUILDFLAG(IS_ANDROID)
   if (base::CommandLine::ForCurrentProcess()->HasSwitch(
